target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-whw01-v1.dts

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "qcom-ipq4019-ap.dk01.1.dtsi"
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Linksys WHW01 v1";
+	compatible = "linksys,whw01-v1";
+
+	aliases {
+		led-boot = &led_system_blue;
+		led-running = &led_system_blue;
+	};
+
+	chosen {
+		bootargs-append = " root=/dev/ubiblock0_0";
+	};
+
+	reserved-memory {
+		rsvd1@87000000 {
+			reg = <0x87000000 0x500000>;
+			no-map;
+		};
+
+		wifi_dump@87500000 {
+			reg = <0x87500000 0x600000>;
+			no-map;
+		};
+
+		rsvd2@87B00000 {
+			reg = <0x87b00000 0x300000>;
+			no-map;
+		};
+	};
+
+	soc {
+		gpio_keys {
+			pinctrl-0 = <0x52>;
+			pinctrl-names = "default";
+			compatible = "gpio-keys";
+
+			button@1 {
+				label = "reset";
+				linux,code = <0x198>;
+				gpios = <&tlmm 0x3f 0x01>;
+				linux,input-type = <0x01>;
+			};
+		};
+	};
+};
+
+&tlmm {
+	i2c_0_pins: i2c_0_pinmux {
+		mux {
+			function = "blsp_i2c0";
+			pins = "gpio58", "gpio59";
+			bias-disable;
+		};
+	};
+
+	reset_pinmux {
+		linux,phandle = <0x52>;
+		phandle = <0x52>;
+
+		mux {
+			pins = "gpio63";
+			bias-pull-up;
+		};
+	};
+};
+
+&spi_0_pins {
+	pinmux_cs {
+		pins = "gpio54", "gpio4";
+	};
+	pinconf_cs {
+		pins = "gpio54", "gpio4";
+	};
+};
+
+&blsp1_spi1 {
+	status = "okay";
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	cs-gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>, <&tlmm 4 GPIO_ACTIVE_HIGH>;
+
+	nor@0 {
+		spi-max-frequency = <0x16e3600>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <0x01>;
+			#size-cells = <0x01>;
+
+			partition@0 {
+				label = "0:SBL1";
+				reg = <0x00 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "0:MIBIB";
+				reg = <0x40000 0x20000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "0:QSEE";
+				reg = <0x60000 0x60000>;
+				read-only;
+			};
+
+			partition@c0000 {
+				label = "0:CDT";
+				reg = <0xc0000 0x10000>;
+				read-only;
+			};
+
+			partition@e0000 {
+				label = "APPSBL";
+				reg = <0xd0000 0xa0000>;
+				read-only;
+			};
+
+			partition@160000 {
+				label = "0:ART";
+				reg = <0x170000 0x10000>;
+				read-only;
+			};
+
+			partition@170000 {
+				label = "u_env";
+				reg = <0x180000 0x40000>;
+			};
+
+			partition@190000 {
+				label = "s_env";
+				reg = <0x1c0000 0x20000>;
+			};
+
+			partition@1b0000 {
+				label = "devinfo";
+				reg = <0x1e0000 0x20000>;
+			};
+		};
+	};
+
+	nand@1 {
+		reg = <1>;
+		status = "ok";
+		compatible = "spi-nand";
+		spi-max-frequency = <0x16e3600>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <0x01>;
+			#size-cells = <0x01>;
+
+			partition@0 {
+				label = "kernel";
+				reg = <0x00 0x5000000>;
+				read-only;
+			};
+
+			partition@600000 {
+				label = "rootfs";
+				reg = <0x600000 0x4a00000>;
```

Comment:
Actually, it would be better to use 7 digits for all values inside reg here to improve readability (for the nand).

Reply:
These `partition` blocks have all just been re-generated from the factory firmware. These should now be correct. Padded to the same number of digits.